ee/server/service/service_teams.go

```diff
@@ -248,8 +248,8 @@ func (svc *Service) ModifyTeamEnrollSecrets(ctx context.Context, teamID uint, se
 	if err := svc.authz.Authorize(ctx, &fleet.EnrollSecret{TeamID: ptr.Uint(teamID)}, fleet.ActionWrite); err != nil {
 		return nil, err
 	}
-	if len(secrets) < 1 {
-		return nil, fleet.NewInvalidArgumentError("secrets", "need to define at least one secret for the team")
+	if secrets == nil {
```

Comment:
If I send `secrets: null` secrets would be nil. It would error here, but it would be a valid way to clear the secrets, right?

Reply:
@chiiph My goal was that the only valid way to clear secrets would be if the secrets payload is an empty array. I tried to document that as a requirement in the API docs too. Maybe I'm overthinking it, but I wanted there to be a higher degree of precision in what it would take to delete all secrets.

Reply:
Sounds good!